core/src/main/java/de/bwaldvogel/mongo/backend/QueryResult.java

```diff
@@ -0,0 +1,43 @@
+package de.bwaldvogel.mongo.backend;
+
+import java.util.Collections;
+import java.util.Iterator;
+
+import de.bwaldvogel.mongo.bson.Document;
+
+public class QueryResult<T extends Document> implements Iterable<T> {
+
+    private Iterable<T> documents;
```

Comment:
As usual adding, adding `final` to all non-modifiable class fields is a good practice.